### PR TITLE
Add necessary files to support running the octane benchmark

### PR DESCRIPTION
**Pull request by CurryKitten**
In support of #75 this PR adds the .js file to run Octane and the .json file to support the chart generation.

Our internal version of Octane had several layers of script wrappers, which I've removed to keep things as simple as possible.  Our .js wrapper also split out the tests individually in order to allow us to bypass certain tests on certain hardware platforms - so I'm using the code that @bnoordhuis pasted in the original issue - with a very small mod.

In order to run this via Jenkins, the process would be -

The normal clone/make for node.js
git clone nodejs/benchmarking
git clone chromium/octane (https://github.com/chromium/octane.git)

In order to make sure we always test on a consistent version of octane I suggest we checkout the (as of now) latest commit, as there's only one release/tag from 2013

`git checkout 4852334fea692f02ce1504f16c625d63abb15183`

Run the benchmark, keeping a log to extract the score from and allowing the metrics to echo to STDOUT so jenkins can capture them, then grab the score with awk

```
node octane | tee octane.log
cat octane.log | awk -F: '/Score/ {print $2}' | sed 's/ //g'
```

### CODE REVIEW

**Comment by gareth-ellis** [comment]
I've not got an environment to hand to check, however if my memory serves Octane throws out individual scores for each of the different tests, rather than one overall score.

We then have a decision to make whether we take a geomean of all the scores (in which case we need to include some calculation of a geomean in the jenkins job or have a wrapper script in the repo), or store each score seperately (in which case we need to add more lines to the octane.json file / or perhaps multipe files).

The skipping on certain hardware platforms was a big endian vs little endian thing, that some tests don't work properly due to binary little endian data in the tests. I'm not sure what the plan going forward is - there was some talk of extra hardware for benchmarking in the community  - if this is going to be exclusively little endian, then we don't need to worry - if there's a chance of there being big endian hardware too, we may need to point out that a geomean is the little endian geomean.

File: benchmarks/octane/octane.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+'use strict';
+const fs = require('fs');
+const vm = require('vm');
+
+global.load = load;
+global.read = read;
+global.print = console.log;
+
+process.chdir('octane')
```
Comment (bnoordhuis) [inline benchmarks/octane/octane.js]: `process.chdir(__dirname)`?

File: benchmarks/octane/octane.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+load('run.js');
+
+function load(filename) {
+  vm.runInThisContext(read(filename));
+}
+
+function read(filename) {
+  return fs.readFileSync(filename, 'utf8');
+}
+
```
Comment (bnoordhuis) [inline benchmarks/octane/octane.js]: Can you drop the blank line?

**Comment by CurryKitten** [comment]
@gareth-ellis Octane does put out a score at the end, which is a Geomean.  Although this did also appear in our jobs, for some reason we were calculating the results manually from the scores of all the individual benchmarks... although the results are at least the same.

File: benchmarks/octane/octane.js (hunk 2 above)
Comment (CurryKitten) [inline benchmarks/octane/octane.js]: @bnoordhuis Yes - have done so

File: benchmarks/octane/octane.js (hunk 1 above)
Comment (CurryKitten) [inline benchmarks/octane/octane.js]: @bnoordhuis The intention is to chdir into the octane directory that had been git cloned - so run.js is in the right place to load all the other tests.  This was based on already being in the benchmarking/benchmarks/octane directory and then cloning octane from there - which is possibly not the best/most desirable course of action, depending on what the normal course of action is in the jenkins scripts

**Comment by gareth-ellis** [comment]
@currykitten , I wonder if that was due to the big endian / little endian calculation....

File: benchmarks/octane/octane.js (hunk 1 above)
Comment (bnoordhuis) [inline benchmarks/octane/octane.js]: Ah, okay.  In that case you may want to write it as `process.chdir(__dirname + '/octane')` to make it independent of the current working directory.

Comment (CurryKitten) [inline benchmarks/octane/octane.js]: @bnoordhuis Good call, thanks - have modified the code.

**Comment by mhdawson** [comment]
@CurryKitten as a note for next time the commit message was longer than the allowed 52 chars.  I shortened it as part of landing.

Landed as: https://github.com/nodejs/benchmarking/commit/15fdf8482bf3ab11aa8d647f95b0480b8d9559bd

**Pull request closed without merging**